tests/hp/prepare_coarsening_and_refinement_01.cc

```diff
@@ -0,0 +1,114 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2021 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+
+// verify restrictions on level differences imposed by
+// DoFHandler::prepare_coarsening_and_refinement()
```

Comment:
I must admit I don't get test. Wouldn't it be more clear if you would loop over all cells and over all it (face) neighbors and compare/output the level difference between each!?

Reply:
Yes, this would've worked, too. I wanted to provide tests that resemble more a use case, i.e., multiple p-refinements.

It just happens that in this scenario, you could 'measure' the p-level of a cell by the distance to the center. Comparing levels of neighboring cells would have worked, too.